### PR TITLE
Add kafka metrics related to local queue size

### PR DESCRIPTION
**Pull request by agbpatro**
# Description

Adds a go routine which periodically tracks pending messages to be sent to the broker.

## Type of change

Please select all options that apply to this change:

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [x] My code follows the style of this project.
- [x] I have performed a self-review of my code.
- [ ] I have made corresponding updates to the documentation.
- [ ] I have added/updated unit tests to cover my changes.
- [ ] I have added/updated integration tests to cover my changes.

### CODE REVIEW

File: datastore/kafka/kafka.go [hunk 1]
```diff
@@ -130,6 +132,17 @@ func (p *Producer) logError(err error) {
 	metricsRegistry.errorCount.Inc(map[string]string{})
 }
 
+func (p *Producer) reportProducerMetrics() {
+	interval := time.Minute
+	t := time.NewTicker(interval)
+	for range t.C {
+		total := p.kafkaProducer.Len()
```
Comment (agbpatro) [inline datastore/kafka/kafka.go:139]: this is total length including the delivered events

File: datastore/kafka/kafka.go [hunk 2]
```diff
@@ -27,11 +27,12 @@ type Producer struct {
...
-	bytesTotal      adapter.Counter
-	produceAckCount adapter.Counter
-	bytesAckTotal   adapter.Counter
-	errorCount      adapter.Counter
+	produceCount     adapter.Counter
+	bytesTotal       adapter.Counter
+	produceAckCount  adapter.Counter
+	bytesAckTotal    adapter.Counter
+	errorCount       adapter.Counter
+	produceQueueSize adapter.Gauge
```
Comment (vmallet) [inline datastore/kafka/kafka.go:35]: `producer`QueueSize?

File: datastore/kafka/kafka.go [hunk 3]
```diff
@@ -130,6 +132,18 @@ func (p *Producer) logError(err error) {
 	metricsRegistry.errorCount.Inc(map[string]string{})
 }
 
+func (p *Producer) reportProducerMetrics() {
+	interval := time.Minute
```
Comment (vmallet) [inline datastore/kafka/kafka.go:136]: One minutes feels like a really long interval; 5s? It should be super cheap right?

File: datastore/kafka/kafka.go (hunk 2 above)
Comment (agbpatro) [reply datastore/kafka/kafka.go]: updated

File: datastore/kafka/kafka.go (hunk 3 above)
Comment (agbpatro) [reply datastore/kafka/kafka.go]: updated

**Review by vmallet — APPROVED** [review]

**Pull request merged**